test/test_datapipe.py

```diff
@@ -586,6 +586,44 @@ def test_concat_datapipe(self):
 
         self.assertEqual(list(concat_dp), list(range(10)) + list(range(5)))
 
+    def test_fork_datapipe(self):
+        input_dp = IDP(range(10))
+
+        # Test case: one child DataPipe yields all value first
+        dp1, dp2, dp3 = input_dp.fork(3)
+        output1, output2, output3 = [], [], []
+        for n1 in dp1:
+            output1.append(n1)
+        for n2 in dp2:
+            output2.append(n2)
+        for n3 in dp3:
+            output3.append(n3)
+        self.assertEqual(output1, list(range(10)))
```

Comment:
nit: If im not mistaken the order is `expected, got`

Reply:
I will switch the order.